### PR TITLE
Fix multiplayer and other bugs

### PR DESCRIPTION
**Pull request by BrianDassow**
Fixes to multiplayer and username labels.

### CODE REVIEW

File: Game.gd [hunk 1]
```diff
@@ -8,16 +8,24 @@ func _ready():
...
 	add_child(new_dungeon)
-	new_dungeon.get_node("walls").add_child(new_player)
 	
-	new_player.name = str(get_tree().get_network_unique_id())
 	new_player.set_network_master(get_tree().get_network_unique_id())
 	
+	new_player.name = str(get_tree().get_network_unique_id())
+	
+	
+	new_dungeon.get_node("walls").add_child(new_player)
```
Comment (MrLane13) [inline Game.gd]: So we add the child to the walls?

File: MainScreen.gd [hunk 2]
```diff
@@ -14,12 +14,12 @@ func _on_joinGameButton_pressed():
 	
 	if userName == "":
 		return
-	Network.connect_to_server(userName)
+		
+	Network.connect_to_server(userName, ip)
```
Comment (MrLane13) [inline MainScreen.gd]: Did the docs not specify an ip before, or how'd you figure out to add it?

File: Network.gd [hunk 3]
```diff
@@ -46,11 +46,13 @@ func _on_player_connected(connected_player_id):
 		rpc_id(1, '_request_player_info', local_player_id, connected_player_id)
 
 remote func _request_player_info(request_from_id, player_id):
+	print("more request?")
```
Comment (MrLane13) [inline Network.gd]: Do we still need these bad boys?

File: Network.gd [hunk 4]
```diff
@@ -59,9 +61,12 @@ remote func _request_players(request_from_id):
 remote func _send_player_info(id, info):
 	players[id] = info
 	var new_player = load('res://Player.tscn').instance()
+	
 	new_player.name = str(id)
 	new_player.set_network_master(id)
-	$'/root/Game/'.add_child(new_player)
+	new_player.get_node("usernameLabel").set_text(info.name)
+	$'/root/Game/dungeon/walls'.add_child(new_player)
```
Comment (MrLane13) [inline Network.gd]: Could this be a variable instead of verbose text?

File: dungeon.tscn [hunk 5]
```diff
@@ -43,7 +43,7 @@ collision_layer = 1
...
...655349, 2, -1200553578, -655348, 2, -1200553578, -655346, 3, -1200553578, -655345, 2, -1200553578, -589813, 2, -1200553578, -589809, 2, -1200553578, -524277, 2, -1200553578, -524273, 2, -1200553578, -458747, 2, -1200553578, -458746, 2, -1200553578, -458745, 2, -1200553578, -458744, 536870916, -1200553578, -458743, 536870916, -1200553578, -458742, 2, -1200553578, -458741, 2, -1200553578, -458740, 2, -1200553578, -458738, 2, -1200553578, -458737, 2, -1200553578, -393211, 2, 0, -393209, 3, -1200553578, -393205, 3, -1200553578, -393201, 2, -1200553578, -327665, 2, -1200553578, -262129, 2, -1200553578, -196603, 2, 0, -196601, 3, 0, -196593, 2, -1200553578, -131067, 2, 0, -131062, 2, -1200553578, -131061, 2, -1200553578, -131060, 2, -1200553578, -131059, 2, -1200553578, -131058, 2, -1200553578, -131057, 2, -1200553578, -65531, 2, 0, -65530, 2, 0, -65527, 2, -1200553578, -65526, 2, -1200553578, 5, 2, -1200553578, 10, 2, 0, 65541, 2, -1200553578, 65546, 2, 0, 131077, 2, -1200553578, 131081, 3, 0, 131082, 2, 0, 196613, 2, -1200553578, 196618, 2, 0, 262149, 2, -1200553578, 262150, 2, -1200553578, 262151, 2, -1200553578, 262152, 2, -1200553578, 262153, 2, -1200553578, 262154, 2, -1200553578 )
```
Comment (MrLane13) [inline dungeon.tscn]: These tscn file diffs scare me. Do you think we can get away with adding them to .gitignore? Or are there still really important changes being made here?

Comment (Ex3Dophos) [reply dungeon.tscn]: I'm okay with this as well. We could look into using something like dropbox to maintain if this is an issue. But i think github is the best moving forward.... as long as we are specific in our commits to what is being changed.

File: Network.gd [hunk 6]
```diff
@@ -46,11 +46,13 @@ func _on_player_connected(connected_player_id):
 		rpc_id(1, '_request_player_info', local_player_id, connected_player_id)
 
 remote func _request_player_info(request_from_id, player_id):
+	print("more request?")
 	if get_tree().is_network_server():
 		rpc_id(request_from_id, '_send_player_info', player_id, players[player_id])
 
 # A function to be used if needed. The purpose is to request all players in the current session.
 remote func _request_players(request_from_id):
+	print("request?")
```
Comment (Ex3Dophos) [inline Network.gd]: probably dont need this either :P

File: Network.gd (hunk 4 above)
Comment (MrLane13) [reply Network.gd]: haha ok :D

**Comment by Ex3Dophos** [comment]
Oh... one more thing. @BrianDassow did you fix the issue with the menu not updating the IP address when you type it in?

**Comment by BrianDassow** [comment]
yes i did

**Pull request merged**